edb/edgeql/compiler/stmt.py

```diff
@@ -132,8 +132,8 @@ def compile_ForQuery(
             iterator_stmt = setgen.new_set_from_set(
                 iterator_view, preserve_scope_ns=True, ctx=scopectx)
 
-            ptr_target = inference.infer_type(iterator_stmt, ctx.env)
-            anytype = ptr_target.find_any(ctx.env.schema)
+            iterator_type = inference.infer_type(iterator_stmt, ctx.env)
```

Comment:
`ir.ast.Set` already has its type inferred, so you can just do `setgen.get_set_type(foo, ctx=ctx)`.  `infer_type` is smart enough to avoid re-inferring the type and is roughly the same, but `get_set_type` is cleaner.